### PR TITLE
feat(delay): create composition functions

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Description
delayable mixin

## Motivation and Context
Vue 3

## How Has This Been Tested?
was not

## Markup:

<details>

```vue
<script lang="ts">
  import { h, defineComponent, ref } from 'vue'
  import { delayProps, useDelay } from '../src/effects/delay'

  export default defineComponent({
    props: {
      ...delayProps({
        openDelay: 300,
        closeDelay: 1000,
      }),
    },
    setup (props) {
      const isActive = ref(false)
      const { runOpenDelay, runCloseDelay } = useDelay(props, value => (isActive.value = value))
      const button = () => h('button', {
        onClick: () => {
          if (isActive.value) {
            // callback
            runCloseDelay()
          } else {
            // promise
            runOpenDelay().then(() => console.log('I am a promise now'))
          }
        },
      }, [
        isActive.value ? 'active' : 'not active',
      ])

      return () => [button()]
    },
  })
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/composables/delay.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export interface DelayProps {
+  closeDelay: number | string
+  openDelay: number | string
+}
+
+export function delayProps (
```
Comment (nekosaur) [inline packages/vuetify/src/composables/delay.ts:6]:
```suggestion
export function makeDelayProps (
```

File: packages/vuetify/src/composables/delay.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export interface DelayProps {
+  closeDelay: number | string
+  openDelay: number | string
```
Comment (nekosaur) [inline packages/vuetify/src/composables/delay.ts:3]:
```suggestion
  closeDelay?: number | string
  openDelay?: number | string
```

**Review by nekosaur — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/composables/__tests__/delay.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { mount } from '@vue/test-utils'
```
Comment (johnleider) [inline packages/vuetify/src/composables/__tests__/delay.spec.ts:1]:
```suggestion
// Utilities
import { mount } from '@vue/test-utils'
```

File: packages/vuetify/src/composables/__tests__/delay.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  it('should cancel delay when running a new one', async () => {
+    const cb = jest.fn()
+    const { runOpenDelay, runCloseDelay } = useDelay({
+      closeDelay: 50,
+      openDelay: 50,
+    }, cb)
+
+    runOpenDelay()
+    await wait(30)
```
Comment (johnleider) [inline packages/vuetify/src/composables/__tests__/delay.spec.ts:65]:
```suggestion
    await wait(30)

```

Comment (johnleider) [reply packages/vuetify/src/composables/__tests__/delay.spec.ts:65]: This is a superfluous suggestion

File: packages/vuetify/src/composables/__tests__/delay.spec.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { mount } from '@vue/test-utils'
+import { delayProps, useDelay } from '../delay'
+import { wait } from '../../../test'
+
```
Comment (johnleider) [inline packages/vuetify/src/composables/__tests__/delay.spec.ts:4]:
```suggestion
// Globals
import { IN_BROWSER } from '../../util/globals'

```

File: packages/vuetify/src/composables/delay.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      type: [Number, String],
+      default: defaults.openDelay,
+    },
+  }
+}
+
+export function useDelay (props: DelayProps, cb?: (value: boolean) => void) {
+  const delays: Partial<Record<keyof DelayProps, number>> = {}
+
+  const runDelayFactory = (prop: keyof DelayProps) => (): Promise<boolean> => {
```
Comment (johnleider) [inline packages/vuetify/src/composables/delay.ts:24]:
```suggestion
  if (!IN_BROWSER) return
  
  const runDelayFactory = (prop: keyof DelayProps) => (): Promise<boolean> => {
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — APPROVED** [review]

**Pull request merged**